File: app/src/main/java/com/example/vknewsfeed/views/LoadingCardView.java
package com.example.vknewsfeed.views;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;

import androidx.annotation.Nullable;
import androidx.cardview.widget.CardView;

import com.example.vknewsfeed.R;

public class LoadingCardView extends CardView {
    private static final int[] STATE_LOADING = {R.attr.state_loading};
    private boolean isLoading = true;

    public LoadingCardView(Context context) {
        super(context);
    }

    public LoadingCardView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public LoadingCardView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected int[] onCreateDrawableState(int extraSpace) {
        if (isLoading) {
            final int[] drawableState = super.onCreateDrawableState(extraSpace + 1);
            mergeDrawableStates(drawableState, STATE_LOADING);
            return drawableState;
        } else {
            return super.onCreateDrawableState(extraSpace);
        }
    }

    public void setLoading(boolean isLoading) {
        if (this.isLoading != isLoading) {
            this.isLoading = isLoading;
            refreshDrawableState();
        }
    }
}

File: app/src/main/java/com/example/vknewsfeed/views/LoadingCardView.kt
package com.example.vknewsfeed.views

import android.content.Context
import android.util.AttributeSet
import androidx.cardview.widget.CardView
import com.example.vknewsfeed.R

class LoadingCardView : CardView {
    private var isLoading = true

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onCreateDrawableState(extraSpace: Int): IntArray {
        return if (isLoading) {
            val drawableState = super.onCreateDrawableState(extraSpace + 1)
            mergeDrawableStates(drawableState, STATE_LOADING)
            drawableState
        } else {
            super.onCreateDrawableState(extraSpace)
        }
    }

    fun setLoading(isLoading: Boolean) {
        if (this.isLoading != isLoading) {
            this.isLoading = isLoading
            refreshDrawableState()
        }
    }

    companion object {
        private val STATE_LOADING = intArrayOf(R.attr.state_loading)
    }
}